buildSrc/src/main/groovy/org/elasticsearch/gradle/vagrant/VagrantCommandTask.groovy

```diff
@@ -32,21 +34,37 @@ import javax.inject.Inject
 public class VagrantCommandTask extends LoggedExec {
 
     @Input
+    String command
+
+    @Input @Optional
+    String subcommand
+
+    @Input
     String boxName
 
     @Input
     Map<String, String> environmentVars
 
     public VagrantCommandTask() {
         executable = 'vagrant'
+    }
 
-        project.afterEvaluate {
-            // It'd be nice if --machine-readable were, well, nice
-            standardOutput = new TeeOutputStream(standardOutput, createLoggerOutputStream())
-            if (environmentVars != null) {
-                environment environmentVars
-            }
+    @Override @TaskAction
+    protected void exec() {
+        // Build our command line for vagrant
+        def vagrantCommand = [executable, command]
+        if (subcommand != null) {
+            vagrantCommand + subcommand
         }
+        commandLine([*vagrantCommand, boxName, *args])
+
+        // It'd be nice if --machine-readable were, well, nice
+        standardOutput = new TeeOutputStream(standardOutput, createLoggerOutputStream())
+        if (environmentVars != null) {
+            environment environmentVars
+        }
+
+        super.exec()
```

Comment:
We should not be overriding and calling the task action directly. Gradle actually has plans to fail if trying to call execute directly in a future version. I think instead you can setup the command line and env vars in a doFirst closure?